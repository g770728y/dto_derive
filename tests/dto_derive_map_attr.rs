use dto_derive::Dto;

#[derive(Debug, PartialEq)]
struct Entity {
    pub field_a: String,
    pub field_b: i32,
    pub field_c: bool,
}

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(map = "field_a: a")]
#[dto(map = "field_b: b")]
#[dto(map = "field_c: c")]
struct DtoRequest {
    a: String,
    b: i32,
    c: bool,
}

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(map = "a: field_a")]
#[dto(map = "b: field_b")]
#[dto(map = "c: field_c")]
struct DtoResponse {
    a: String,
    b: i32,
    c: bool,
}

#[test]
fn dto_map_request() {
    let request = DtoRequest {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let expected = Entity {
        field_a: "test_simple".to_string(),
        field_b: 123i32,
        field_c: true,
    };

    let actual: Entity = request.into();
    assert_eq!(expected, actual);
}

#[test]
fn dto_map_response() {
    let entity = Entity {
        field_a: "test_simple".to_string(),
        field_b: 123i32,
        field_c: true,
    };

    let expected = DtoResponse {
        a: "test_simple".to_string(),
        b: 123i32,
        c: true,
    };

    let actual: DtoResponse = entity.into();
    assert_eq!(expected, actual);
}
