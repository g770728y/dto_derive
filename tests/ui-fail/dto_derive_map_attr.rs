extern crate dto_derive;
use dto_derive::Dto;

#[derive(Debug, PartialEq, Dto)]
#[dto(map = "a: b")]
#[dto(map = "a: c")] //~ ERROR could not map already mapped field 'a'
struct Dto1 { }

fn main() { }
