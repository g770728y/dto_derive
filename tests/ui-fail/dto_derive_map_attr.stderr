error: could not map already mapped field 'a'
 --> $DIR/dto_derive_map_attr.rs:6:6
  |
6 | #[dto(map = "a: c")] //~ ERROR could not map already mapped field 'a'
  |      ^^^^^^^^^^^^^^

error: aborting due to previous error

